task/08-objects-tasks.js

```diff
@@ -107,40 +107,82 @@ function fromJSON(proto, json) {
  */
 
 const cssSelectorBuilder = {
-
-    element: function(value) {
-        throw new Error('Not implemented');
-    },
-
-    id: function(value) {
-        throw new Error('Not implemented');
-    },
-
-    class: function(value) {
-        throw new Error('Not implemented');
-    },
-
-    attr: function(value) {
-        throw new Error('Not implemented');
-    },
-
-    pseudoClass: function(value) {
-        throw new Error('Not implemented');
-    },
-
-    pseudoElement: function(value) {
-        throw new Error('Not implemented');
-    },
-
-    combine: function(selector1, combinator, selector2) {
-        throw new Error('Not implemented');
-    },
+  selector: "",
+
+  element(value) {
+    const obj = Object.create(cssSelectorBuilder);
+    obj.key = 1;
+    this.error(obj.key);
+    obj.selector = this.selector + value;
+    return obj;
+  },
+
+  id(value) {
+    const obj = Object.create(cssSelectorBuilder);
+    obj.key = 2;
+    this.error(obj.key);
+    obj.selector = this.selector + "#" + value;
+    return obj;
+  },
+
+  class(value) {
+    const obj = Object.create(cssSelectorBuilder);
+    obj.key = 3;
+    this.error(obj.key);
+    obj.selector = this.selector + "." + value;
+    return obj;
+  },
+
+  attr(value) {
+    const obj = Object.create(cssSelectorBuilder);
+    obj.key = 4;
+    this.error(obj.key);
+    obj.selector = this.selector + "[" + value + "]";
+    return obj;
+  },
+
+  pseudoClass(value) {
+    const obj = Object.create(cssSelectorBuilder);
+    obj.key = 5;
+    this.error(obj.key);
+    obj.selector = this.selector + ":" + value;
+    return obj;
+  },
+
+  pseudoElement(value) {
+    const obj = Object.create(cssSelectorBuilder);
+    obj.key = 6;
+    this.error(obj.key);
+    obj.selector = this.selector + "::" + value;
+    return obj;
+  },
+
+  combine(selector1, combinator, selector2) {
+    const obj = Object.create(cssSelectorBuilder);
+    obj.selector =
+      selector1.selector + " " + combinator + " " + selector2.selector;
+    return obj;
+  },
+
+  stringify() {
+    return this.selector;
+  },
+
+  error(key) {
+    if (this.key > key)
+      throw new Error(
```

Comment:
If the if statement is too long for one line and the prettier breaks it into two lines, please wrap it with `{ }`